## <font color='blue'>**Sachin Tendulkar ODI records for India**</font>

In [ ]:
!wget --no-check-certificate https://drive.google.com/uc?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7 -O Sachin_ODI.csv

--2024-10-02 16:22:57--  https://drive.google.com/uc?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7
Resolving drive.google.com (drive.google.com)... 173.194.216.101, 173.194.216.100, 173.194.216.139, ...
Connecting to drive.google.com (drive.google.com)|173.194.216.101|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://drive.usercontent.google.com/download?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7 [following]
--2024-10-02 16:22:57--  https://drive.usercontent.google.com/download?id=1zBM3idCNWceBMLKMRBTN8aVFAg5U_YH7
Resolving drive.usercontent.google.com (drive.usercontent.google.com)... 172.217.203.132, 2607:f8b0:400c:c07::84
Connecting to drive.usercontent.google.com (drive.usercontent.google.com)|172.217.203.132|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 26440 (26K) [application/octet-stream]
Saving to: ‘Sachin_ODI.csv’

Sachin_ODI.csv      100%[===================>]  25.82K  --.-KB/s    in 0.001s  

2024-10-02 16:23:00 (48.5 MB

In [ ]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [ ]:
df_sachin = pd.read_csv("Sachin_ODI.csv")

In [ ]:
df_sachin.head()

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

In [ ]:
# shape of the dataset

df_sachin.shape

(360, 14)

### <font color='purple'> **A match is randomly chosen, what is the probability that India have won that match?**</font>

In [ ]:
# find the rows where India have won and store into new dataframe
df_won=df_sachin.loc[df_sachin["Won"]==True]

In [ ]:
# calculate the number of True values which is our possible outcome
df_won.shape[0]

184

In [ ]:
prob_winning=len(df_won)/len(df_sachin)
prob_winning

0.5111111111111111

<font color='orange'> Conclusion: </font>:  

If a match is randomly chosen, there is **51%** chance that India have won that match.

### <font color='purple'>  **A match is chosen at a random, what is the probability that Sachin has scored a Century in that match?** </font>   



In [ ]:
# using value_counts()

df_sachin["century"].value_counts()

century
False    314
True      46
Name: count, dtype: int64

In [ ]:
46/360

0.12777777777777777

<font color='orange'> Conclusion: </font>
  
  If you chose a random match, there is **12.77% chance** that Sachin has scored a century in that match


<font color='orange'> Let's find out how many matches India have won when Sachin has **`scored a century`** and  </font>

<font color='orange'> How many matches India have won when sachin **`didn't score a century`**.  </font>    




In [ ]:
df_sachin[["century","Won"]].value_counts()

century  Won  
False    False    160
         True     154
True     True      30
         False     16
Name: count, dtype: int64

### <font color='purple'>**Cross Tab and contingency table**</font>

In [ ]:
pd.crosstab(index=df_sachin["century"],
            columns=df_sachin["Won"],
            margins=True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

### <font color='purple'> **A match is chosen at a random. What is the probability that Sachin has scored a century in that match and India have won that match?**</font>

In [ ]:
prob_sachin_century_won=30/360
prob_sachin_century_won

0.08333333333333333

<font color='orange'> Conclusion : </font>
  
  There is **8% chance** that Sachin has scored a century and India have won that match if we choose a random match

### <font color='purple'>**Conclusion of the Problem statement:** </font>    

Let's have a look how is Sachin's batting can or cannot impact the winning chances of India  


1. Out of the <font color='purple'>**360** matches</font> that Sachin has played, <font color='purple'>India have **won 184** matches and Loose 176 matches. </font>

2. So, if we choose any match at a random from Sachin's ODI career, there is a <font color='purple'>**51%** chance that India have won that match.     </font>


3. Now, If we choose a random match from Sachin's ODI career, there is <font color='purple'>**12.77%** chance that Sachin has scored a century in that match.  </font>

4. We know if a random match is choosen, there is 12.77% chance that Sachin has scored a century but there is**only 8%** chance India have won that match.

  - we can conclude that the <font color='orange'>chances of India, Winning a match is more when Sachin didn't score a century </font>





<br>

Finally,

We can conclude that, if we pick a random match where Sachin played, India's win percentage is 51%. There is 12.77% chance of Sachin scoring a century in that match, and there is only 8% chance that in that match Sachin scores a century as well as India have won that match


<br><br>

# <font color='purple'>**Sachin Tendulkar batting for India**</font>

# Marginal Probability
**What is the probability that Sachin's team wins the match**

In [ ]:

#𝑃_𝑊 =𝑁𝑜𝑜𝑓𝑚𝑎𝑡𝑐ℎ𝑒𝑠𝑤𝑜𝑛𝑏𝑦𝑆𝑎𝑐ℎ𝑖𝑛/𝑇𝑜𝑡𝑎𝑙𝑛𝑜𝑜𝑓𝑚𝑎𝑡𝑐ℎ𝑒𝑠

184/360


0.5111111111111111

**What is the probability of Sachin scoring a century**

In [ ]:
#𝑃(𝐶) = =𝑁𝑜 𝑜𝑓 𝑚𝑎𝑡𝑐ℎ𝑒𝑠 𝑤𝑖𝑡ℎ 𝑐𝑒𝑛𝑡𝑢𝑟𝑦/𝑇𝑜𝑡𝑎𝑙 𝑛𝑜 𝑜𝑓 𝑚𝑎𝑡𝑐ℎ𝑒𝑠
46/360


0.12777777777777777

**probability that Sachin's team wins AND he scores a century**

In [ ]:
pd.crosstab(index=df_sachin["century"],
            columns=df_sachin["Won"],
            margins=True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

In [ ]:
30/360


0.08333333333333333

**probability that Sachin scores a century AND his team loses**

In [ ]:
16/360

0.044444444444444446

#  **Here we calculated the likelihood of two events occurring together and at the same point in time.This type of probability value is known as Joint Probability**

# probability that Sachin's team wins the match given that he scored a century

In [ ]:
#p(Wins teh match|Given he scored century)
30/46

0.6521739130434783

# probability that Sachin scores a century, given that his team has won the match

In [ ]:
#p(Sachin scores a century|iven that his team has won the match)
30/184

0.16304347826086957